File: project-001/bank-collection-service/internal/domain/model/user_bank_list.go
package model

import (
	"time"
)

type UserBankList struct {
	ID                uint      `json:"id" gorm:"primaryKey"`
	BankCode          string    `json:"bank_code"`
	BankName          string    `json:"bank_name"`
	AccountHolderName string    `json:"account_holder_name"`
	AccountNumber     string    `json:"account_number" gorm:"unique;not null"`
	UserID            uint      `json:"user_id"`
	CreatedOn         time.Time `json:"created_on" gorm:"autoCreateTime"`
	UpdatedOn         time.Time `json:"updated_on" gorm:"autoUpdateTime"`
}

File: resiliences-service/Models/UserBankList.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace resiliences_service.Models
{
    [Table("UserBankLists")]
    [Index(nameof(AccountNumber), IsUnique = true)]
    public class UserBankList
    {
        [Key]
        public uint Id { get; set; }

        public string? BankCode { get; set; }

        public string? BankName { get; set; }

        public string? AccountHolderName { get; set; }

        [Required]
        public string AccountNumber { get; set; } = default!;

        public uint UserId { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime CreatedOn { get; set; } = DateTime.UtcNow;

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime UpdatedOn { get; set; } = DateTime.UtcNow;
    }
}